plutarch-benchmark/src/Plutarch/Benchmark.hs

```diff
@@ -65,42 +66,46 @@ import qualified Plutus.V1.Ledger.Api as Plutus
 benchmarkScript :: HasCallStack => String -> Script -> NamedBenchmark
 benchmarkScript name = NamedBenchmark . (name,) . benchmarkScript'
 
+{-# DEPRECATED benchmarkScript' "use `Plutarch.Evaluate.evalScript` and `mkBenchmark` with `scriptSize`" #-}
 benchmarkScript' :: HasCallStack => Script -> Benchmark
-benchmarkScript' =
-  uncurry mkBenchmark . (evalScriptCounting &&& (fromInteger . toInteger . SBS.length)) . serialiseScriptShort
+benchmarkScript' script =
+  mkBenchmark
+    (evalScriptCounting $ serialiseScriptShort script)
+    (scriptSize script)
   where
-    mkBenchmark :: Maybe ExBudget -> Int64 -> Benchmark
-    mkBenchmark Nothing =
-      Benchmark Nothing Nothing . ScriptSizeBytes
-    mkBenchmark (Just (ExBudget cpu mem)) =
-      Benchmark (Just cpu) (Just mem) . ScriptSizeBytes
-
-    serialiseScriptShort :: Script -> SBS.ShortByteString
-    serialiseScriptShort = SBS.toShort . LB.toStrict . serialise -- Using `flat` here breaks `evalScriptCounting`
-    evalScriptCounting :: HasCallStack => Plutus.SerializedScript -> Maybe Plutus.ExBudget
+    evalScriptCounting :: HasCallStack => Plutus.SerializedScript -> Plutus.ExBudget
     evalScriptCounting script =
       let costModel = fromJust Plutus.defaultCostModelParams
           (_logout, e) = Plutus.evaluateScriptCounting Plutus.Verbose costModel script []
        in case e of
-            Left _evalErr -> Nothing
-            Right exbudget -> Just exbudget
+            Left evalErr -> error $ show evalErr
+            Right exbudget -> exbudget
 
 data Benchmark = Benchmark
-  { exBudgetCPU :: Maybe ExCPU
-  -- ^ CPU budget used by the script. Nothing if script errors.
-  , exBudgetMemory :: Maybe ExMemory
-  -- ^ Memory budget used by the script. Nothing if script errors.
+  { exBudgetCPU :: ExCPU
+  -- ^ CPU budget used by the script.
+  , exBudgetMemory :: ExMemory
+  -- ^ Memory budget used by the script.
   , scriptSizeBytes :: ScriptSizeBytes
   -- ^ Size of Plutus script in bytes
   }
   deriving stock (Show, Generic)
   deriving anyclass (ToJSON)
 
+mkBenchmark :: ExBudget -> ScriptSizeBytes -> Benchmark
+mkBenchmark (ExBudget cpu mem) = Benchmark cpu mem
+
 newtype ScriptSizeBytes = ScriptSizeBytes Int64
   deriving stock (Eq, Ord, Show, Generic)
   deriving newtype (Num, ToField)
   deriving newtype (ToJSON)
 
+scriptSize :: Script -> ScriptSizeBytes
+scriptSize = ScriptSizeBytes . fromInteger . toInteger . SBS.length . serialiseScriptShort
+
+serialiseScriptShort :: Script -> SBS.ShortByteString
+serialiseScriptShort = SBS.toShort . LB.toStrict . serialise -- Using `flat` here breaks `evalScriptCounting`
```

Comment:
Can you elaborate on the comment BTW?

Reply:
This looks to be an outdated comment; I remember trying to use the `flat` library for some reason, but it is not used anymore. Removed.